test_runner/regress/test_auth.py

```diff
@@ -82,6 +84,115 @@ def test_compute_auth_to_pageserver(neon_env_builder: NeonEnvBuilder):
             assert cur.fetchone() == (5000050000,)
 
 
+def test_pageserver_multiple_keys(neon_env_builder: NeonEnvBuilder):
+    neon_env_builder.auth_enabled = True
+    env = neon_env_builder.init_start()
+
+    pageserver_token_old = env.auth_keys.generate_pageserver_token()
+    pageserver_http_client_old = env.pageserver.http_client(pageserver_token_old)
+
+    pageserver_http_client_old.reload_auth_validation_keys()
+
+    env.auth_public_key_path = (
+        Path(env.repo_dir) / "auth_public_key.pem" / "auth_public_key_old.pem"
+    )
```

Comment:
You set this to match the path of the "old" key, but later when you regenerate the token, you don't change this path. So after regenerating the token, the AuthKeys has a mix of old public key path and new private key path.

I think that's harmless because the AuthKeys.pub path is not actually used for anything, but seems confusing.

Reply:
I've removed the public key part of `AuthKeys`.